src/lib/App/RequestAppResponseRenderer.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace EzSystems\HybridPlatformUi\App;
+
+use EzSystems\HybridPlatformUi\Components\App;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+
+class RequestAppResponseRenderer implements AppResponseRenderer
+{
+    /**
+     * @var \Symfony\Component\HttpFoundation\RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var \Symfony\Component\HttpFoundation\RequestMatcherInterface
+     */
+    private $ajaxUpdateRequestMatcher;
+
+    public function __construct(
+        RequestStack $requestStack,
+        RequestMatcherInterface $ajaxUpdateRequestMatcher
+    ) {
+
+        $this->requestStack = $requestStack;
+        $this->ajaxUpdateRequestMatcher = $ajaxUpdateRequestMatcher;
+    }
+
+    public function render(Response $response, App $app)
+    {
+        $app->setConfig(['mainContent' => ['result' => $response->getContent()]]);
+        $this->configureToolbars($app);
+
+        $appResponse = $this->ajaxUpdateRequestMatcher->matches($this->requestStack->getMasterRequest())
```

Comment:
In which other files does this apply ? As far as I can tell, this is the only injection of `@master_request`; the other cases are EventSubscribers that get the Request from the Event, afaict.